src/KingSystem/World/worldMgr.h

```diff
@@ -0,0 +1,83 @@
+#pragma once
+
+#include <container/seadPtrArray.h>
+#include <mc/seadJobQueue.h>
+#include "KingSystem/Resource/resHandle.h"
+#include "KingSystem/Utils/ParamIO.h"
+#include "agl/Utils/aglAtomicPtrArray.h"
+#include "agl/Utils/aglParameter.h"
+#include "agl/Utils/aglParameterObj.h"
+
+namespace ksys::world {
+
+class Climate {};
+
+class WorldInfo : public ParamIO {
+public:
+    WorldInfo() : ParamIO("winfo", 0) {}
+    ~WorldInfo();
+
+    bool ParamIO_m0() override { return true; }
+
+    res::Handle mResHandle;
+    u32 mNumClimates = 0;
+    Climate* mClimates = nullptr;
+};
+
+class DungeonEnv : public ParamIO {
+public:
+    DungeonEnv() : ParamIO("dgenv", 0) {}
+    ~DungeonEnv();
+
+    bool ParamIO_m0() override { return true; }
+
+    res::Handle mResHandle;
+
+    agl::utl::Parameter<f32> mLightLongitude;
+    agl::utl::Parameter<sead::FixedSafeString<32>> mDungeonSize;
+    agl::utl::Parameter<sead::FixedSafeString<32>> mString538;
+    agl::utl::ParameterObj mDungeonEnvObj;
+    sead::PtrArray<void*> mgrs{};
+    sead::DirectResource* infoRes = nullptr;
+};
+
+// FIXME: incomplete
+class WorldMgr : sead::hostio::Node {
+    WorldMgr();
+
+    WorldInfo mWorldInfo;
+    DungeonEnv mDungeonEnv;
+
+    agl::utl::AtomicPtrArray<void*> mAtomicPtrArray{0, 0};
+    u32 field_5E0 = 1;
```

Comment:
```suggestion
    u32 _5e0 = 1;
```
Fields with dummy names should be named `_{:x}` (underscore + offset in hexadecimal **in lowercase**)